test/Passes/pass-convert-linalg-to-func.mlir

```diff
@@ -0,0 +1,98 @@
+// CHECK: %[[PTR_ARG1:.+]] = memref.extract_aligned_pointer_as_index %[[ARG1]] : memref<?x32xf32> -> index
+// CHECK: %[[PTR_CAST_ARG1:.+]] = arith.index_cast %[[PTR_ARG1]] : index to i64
+// CHECK: %[[LLVM_PTR_ARG1:.+]] = llvm.inttoptr %[[PTR_CAST_ARG1]] : i64 to !llvm.ptr<f32>
+// CHECK: %[[PTR_ARG2:.+]] = memref.extract_aligned_pointer_as_index %[[ARG2]] : memref<64x32xf32> -> index
+// CHECK: %[[PTR_CAST_ARG2:.+]] = arith.index_cast %[[PTR_ARG2]] : index to i64
+// CHECK: %[[LLVM_PTR_ARG2:.+]] = llvm.inttoptr %[[PTR_CAST_ARG2]] : i64 to !llvm.ptr<f32>
+// CHECK: call @linalg_matmul_blas(%[[C64]], %[[C32]], %[[DIM]], %[[LLVM_PTR_ARG0]], %[[C0]], %[[C64]], %[[LLVM_PTR_ARG1]], %[[C0]], %[[DIM]], %[[LLVM_PTR_ARG2]], %[[C0]], %[[C64]])
+
+// -----
+
+func.func @strided_memref(%arg0: memref<64x64xf32, strided<[64, 1], offset: ?>>,
+                          %arg1: memref<64x64xf32>, %arg2: memref<64x64xf32>) {
+  linalg.matmul ins(%arg0, %arg1 : memref<64x64xf32, strided<[64, 1], offset: ?>>, memref<64x64xf32>)
+                outs(%arg2 : memref<64x64xf32>)
+  return
+}
+
+// CHECK-LABEL: strided_memref
+// CHECK: linalg.matmul
+// CHECK-NOT: call @linalg_matmul_blas
+
+// -----
+
+func.func @static_matmul_2(%arg0: memref<3x4xf32>, %arg1: memref<4x5xf32>, %arg2: memref<3x5xf32>) {
+  linalg.matmul ins(%arg0, %arg1 : memref<3x4xf32>, memref<4x5xf32>)
+                outs(%arg2 : memref<3x5xf32>)
+  return
+}
+
+// CHECK-LABEL: static_matmul_2
+// CHECK-SAME: %[[ARG0:.+]]: memref<3x4xf32>, %[[ARG1:.+]]: memref<4x5xf32>, %[[ARG2:.+]]: memref<3x5xf32>
+// CHECK-DAG: %[[C3:.+]] = arith.constant 3 : index
+// CHECK-DAG: %[[C5:.+]] = arith.constant 5 : index
+// CHECK-DAG: %[[C4:.+]] = arith.constant 4 : index
+// CHECK-DAG: %[[C0:.+]] = arith.constant 0 : index
+// CHECK: %[[ARG0_PTR:.+]] = memref.extract_aligned_pointer_as_index %[[ARG0]] : memref<3x4xf32> -> index
+// CHECK: %[[ARG0_PTR_CAST:.+]] = arith.index_cast %[[ARG0_PTR]] : index to i64
+// CHECK: %[[ARG0_LLVM_PTR:.+]] = llvm.inttoptr %[[ARG0_PTR_CAST]] : i64 to !llvm.ptr<f32>
+// CHECK: %[[ARG1_PTR:.+]] = memref.extract_aligned_pointer_as_index %[[ARG1]] : memref<4x5xf32> -> index
+// CHECK: %[[ARG1_PTR_CAST:.+]] = arith.index_cast %[[ARG1_PTR]] : index to i64
+// CHECK: %[[ARG1_LLVM_PTR:.+]] = llvm.inttoptr %[[ARG1_PTR_CAST]] : i64 to !llvm.ptr<f32>
+// CHECK: %[[ARG2_PTR:.+]] = memref.extract_aligned_pointer_as_index %[[ARG2]] : memref<3x5xf32> -> index
+// CHECK: %[[ARG2_PTR_CAST:.+]] = arith.index_cast %[[ARG2_PTR]] : index to i64
+// CHECK: %[[ARG2_LLVM_PTR:.+]] = llvm.inttoptr %[[ARG2_PTR_CAST]] : i64 to !llvm.ptr<f32>
+// CHECK: call @linalg_matmul_blas(%[[C3]], %[[C5]], %[[C4]], %[[ARG0_LLVM_PTR]], %[[C0]], %[[C3]], %[[ARG1_LLVM_PTR]], %[[C0]], %[[C4]], %[[ARG2_LLVM_PTR]], %[[C0]], %[[C3]])
+
+// -----
+
+func.func @static_matmul_double(%arg0: memref<3x4xf64>, %arg1: memref<4x5xf64>, %arg2: memref<3x5xf64>) {
```

Comment:
OneDNN is good for very large kernels. Would be nice to have some examples from `matmul` into `sgemm` calls for large (> 2k x 2k) matrices. We can add that as a small benchmark on a later PR just to see that it's good perf, but if you can just run locally and check, that'd be ok for now.

Reply:
added.